apps/advisor/pkg/app/checktyperegisterer/checktyperegisterer.go

```diff
@@ -73,8 +73,8 @@ func (r *Runner) createOrUpdate(ctx context.Context, log logging.Logger, obj res
 			}
 			currentAnnotations := current.GetAnnotations()
 			annotations := obj.GetAnnotations()
-			maps.Copy(annotations, currentAnnotations)
-			obj.SetAnnotations(annotations)
+			maps.Copy(currentAnnotations, annotations)
+			obj.SetAnnotations(currentAnnotations)
```

Comment:
this change is to allowing changing things like the name annotation, if not the current name will always persist.